src/dataflow/src/sink/kafka.rs

```diff
@@ -176,7 +184,21 @@ pub fn kafka<G>(
         &name.clone(),
         |info| {
             let activator = stream.scope().activator_for(&info.address[..]);
-            move |input| {
+            let shutdown_button = ShutdownButton::new(
+                producer.clone(),
+                stream.scope().activator_for(&info.address[..]),
+            );
+
+            let ret = move |input: &mut FrontieredInputHandle<_, _, _>| {
+                let producer: &RefCell<_> = producer.borrow();
+                let producer = &*producer.borrow();
+                if producer.is_none() {
+                    // The sink has already been dropped
+                    return false;
+                }
+
+                let producer = producer.as_ref().expect("producer known to exist");
```

Comment:
Btw, the Rusty pattern for pattern-matching early returns is to use a match statement with the return inside:

```rust
let producer = match &*producer.borrow() {
    Some(producer) => producer,
    None => return false,
};
```